tools/adventure-pack/goodies/typescript/traverseLevelOrder/index.ts

```diff
@@ -0,0 +1,23 @@
+export function* traverseLevelOrder<
+  T extends { left?: T | null | undefined; right?: T | null | undefined },
+>(root: T | null | undefined): Generator<T[], void, void> {
+  if (!root) {
+    yield [];
+  }
+
+  const queue = [root];
```

Comment:
We can probably merge the queue and level variables into one, and reassign it each time, since the previous level is the queue for the next level.

This is also a good opportunity to use `flatMap`.

Reply:
ack!